02-wrangling.Rmd

```diff
@@ -304,6 +317,76 @@ From this visualization, we can see that in Calgary, Edmonton, Toronto and Vanco
+```
+### Using `filter` to extract rows with values above or below a threshold
+
+If we are interested in finding the languages with higher numbers of people who speak it as their mother tongue than Mandarin---which is  `r format(mandarin_mother_tongue, scientific = FALSE, big.mark = ",")` people ---then we can create a filter to obtain rows where the value of `mother_tongue` is greater than `r format(mandarin_mother_tongue, scientific = FALSE, big.mark = ",")`. In this case, we see that `filter` returns a data frame with two rows; this indicates that two languages are spoken more often as their mother tongue compared to Mandarin.
+
+```{r}
+spoke_often_at_home <- filter(can_lang, mother_tongue > 592040)
+spoke_often_at_home
+```
+
+
+### Using `mutate` to add columns or overwrite existing columns
+
+We found that `r format(mandarin_mother_tongue, scientific = FALSE, big.mark = ",")` 
+people reported that their mother tongue was Mandarin in the 2016 Canadian census. What does this number mean to us? To understand this number, we need context. In particular, how many people were in Canada when this data was collected? From the 2016 Canadian census profile, the population was reported to be `r format(census_popn, scientific = FALSE, big.mark = ",")` people. The number of people who report that Mandarin is their mother tongue is much more meaningful when we report it in this context. We can even go a step further and transform this count to a relative frequency or proportion. We can do this by dividing the number of people reporting a given language as their mother tongue by the number of people who live in Canada. For example, the proportion of people who reported that their mother tongue was Mandarin in the 2016 Canadian census was `r format(round(mandarin_mother_tongue/census_popn, 2), scientific = FALSE, big.mark = ",")`.
+
+
+We can use the `mutate` function in R to calculate the proportion for all of the languages in the 2016 Canadian census data set. `mutate` is useful for creating new columns in a data frame or transforming existing columns. Its general syntax is: 
+
+```{r img-mutate, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Syntax for the mutate function", out.width="1100", fig.retina = 2}
+knitr::include_graphics("img/mutate_function.jpeg")
+```
+
+Below we use `mutate` to calculate the proportion of people reporting a given language as their mother tongue for all the languages in the `can_lang` data set as shown in figure \@ref(fig:img-mutate):
+```{r}
+mutated_can_lang <- mutate(can_lang, mother_tongue_proportion = mother_tongue / 35151728)
+```
+
+Here we used a new data frame name (`mutated_can_lang`) and column name (`mother_tongue_proportion`) to create a new data frame and a new column. But we could have used the existing data frame name (`can_lang`) and column name (`mother_tongue`), which would have written over our original data frame and column.
+
+
 ## Combining functions using the pipe operator, `%>%`:
```

Comment:
We need to move this to the new `|>` pipe operator, but still introduce `%>%` from `magrittr`. Linking to the issue where we discussed how to do this: https://github.com/UBC-DSCI/introduction-to-datascience/issues/153